community/src/community.cpp

```diff
@@ -1503,89 +1547,191 @@ ACTION community::createbadge(
         check(false, "ERR::BADGE_ISSUE_TYPE_INVALID::Badge issue type is invalid");
     }
 
+    auto getByCodeReferId = _codes.get_index<"by.refer.id"_n>();
+    auto issue_badge_code_itr = getByCodeReferId.find(badge_id);
+    uint64_t issue_badge_code_id;
+
+    // find issue badge code of this configuring badge
+    while (issue_badge_code_itr != getByCodeReferId.end()) {
+        if (issue_badge_code_itr->code_type.type == CodeTypeEnum::BADGE && 
+            ( issue_badge_code_itr->code_name == BA_Issue || issue_badge_code_itr->code_name == BA_Claim)) {
+                break;
+        }
+        issue_badge_code_itr++;
+    }
+
```

Comment:
I think you should make combination for the index key for easy searching instead of use this spin code. too long and bad perfomance

Reply:
TABLE codef
    {
        uint64_t code_id;
        name code_name;
        name contract_name;
        vector<name> code_actions;
        uint8_t code_exec_type = ExecutionType::SOLE_DECISION;
        uint8_t amendment_exec_type = ExecutionType::SOLE_DECISION;
        CodeType code_type;

        uint64_t primary_key() const { return code_id; }
        uint64_t by_code_name() const { return code_name.value; }
        uint128_t by_reference_id() const { return (code_type.refer_id || code_type.type << 64; }
    };

    typedef eosio::multi_index<"codes"_n, codef, 
        indexed_by< "by.code.name"_n, const_mem_fun<codef, uint64_t, &codef::by_code_name>>,
        indexed_by< "by.refer.id"_n, const_mem_fun<codef, uint128_t, &codef::by_reference_id>>
        > code_table;